### DISCUSSION TITLE
Between and Within Subject Design with Stimulus Coding

### DISCUSSION THREAD

**Discussion by shabnamhossein** [lnccbrown/HSSM#601] [Q&A]
Hello,

I want to use hierarchical DDM to model an implicit association task (IAT). I have both within (Time: pre and post a drug administration) and between subject (Group: control group vs treatment group) variables. My understanding is that  hddm.HDDMRegressor or hddm.HDDM are not the best models for this purpose, especially since I need to use stimulus coding and not accuracy coding. So I am using the HSSM package and this my model (only considering v for the purpose of this question) including group specific intercepts and slopes and group specific intercept for different time point. 1) Is this a correct formulation? Would you recommend formulating it differently?

model = hssm.HSSM(
	model="ddm”,
	data=data,
	include=[
		{
			"name": "v",
            		"formula": "v ~ Group + (Group|subject) + (1|Time)”,
			“link": "identity",
		},
   		 	],
)

And 2) How can I incorporate stimulus coding with this formulation (v and -v)?

**Comment by AlexanderFengler** [comment]
Hi @shabnamhossein,

sorry for the late response here. 

1. Yes HSSM allows mixing within and between subject variables so your formulation looks good to me. 
2. I previously included stimulus coding via a (1, -1) covariate that indicates the trial type e.g.. An alternative way would be to pass such a column directly in the likelihood and slightly change the likelihood formulation (this is how HDDM had done it). Nothing really prevents us from doing that, but given resource constraints, it's currently not a top priority. If you are interested in contributing please let us know we will help you get set up. 

Best,
Alex
